pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -430,6 +442,53 @@ func (r *PyTorchJobReconciler) UpdateJobStatus(job interface{},
 	return nil
 }
 
+// IsWorker0Completed returns true if pod of worker0 succeeded and exited with 0
+func (p *PyTorchJobReconciler) IsWorker0Completed(job *pytorchv1.PyTorchJob,
+	replicas map[commonv1.ReplicaType]*commonv1.ReplicaSpec) (bool, error) {
+	worker0Completed := false
+	_, ok := replicas[pytorchv1.PyTorchReplicaTypeWorker]
+	if !ok {
+		return true, nil
+	}
+	podSlices, err := p.getPodSlices(job,
```

Comment:
I understand the slice approach definitely works. But what is the point to create a the `getPodSlices` method while all we need is just the worker pod with index=0? And what kind of disadvantages will it have to use `Get` function or `List` with label selector to catch worker0 pod?

Reply:
It is to keep consistency with TF controller. The code is copied from it.